### ISSUE TITLE
improve detection performance

### ISSUE THREAD

**Issue by xiaohai12** [AlexeyAB/darknet#3070]
Hi, I met a problem that my model detected the object depend greatly on object's background color. For example,  my model is likely to be confused by two classes with same background color.
So, is there any solution to solve this problem?

**Comment by AlexeyAB** [comment]
@xiaohai12 Hi,

Can you show examples?

**Comment by xiaohai12** [comment]
![image](https://user-images.githubusercontent.com/27256068/56902497-b70b7500-6a9a-11e9-8227-78a610d82b8f.png)
![image](https://user-images.githubusercontent.com/27256068/56902485-aeb33a00-6a9a-11e9-847b-b92075bd5e6d.png)
The logo in first image was correctly detected but incorrectly detected in second image.
I thought the reason is that these two logos have same background.

**Comment by AlexeyAB** [comment]
> I thought the reason is that these two objects have same background.

Yes, in a few day I will add `blur=1` parameter to the `[net]` section, that will randomly blur the background during training, so it can **partially** solve this problem.

But do you have enough training examples of this logos? ![image](https://user-images.githubusercontent.com/27256068/56902485-aeb33a00-6a9a-11e9-847b-b92075bd5e6d.png)

**Comment by xiaohai12** [comment]
> > I thought the reason is that these two objects have same background.
> 
> Yes, in a few day I will add `blur=1` parameter to the `[net]` section, that will randomly blur the background during training, so it can **partially** solve this problem.
> 
> But do you have enough training examples of this logos? ![image](https://user-images.githubusercontent.com/27256068/56902485-aeb33a00-6a9a-11e9-847b-b92075bd5e6d.png)

Actually,  this 'logo' is not considered as logo in my dataset, so it should not be detected .

**Comment by xiaohai12** [comment]
Btw, I already have some blurred data in my dataset. 
I will try to set **blur=1** to see the result.

**Comment by AlexeyAB** [comment]
> Btw, I already have some blurred data in my dataset.

The entire image, except for objects, should be very blurry.

> I will try to set blur=1 to see the result.

I will implement it in a few days.

**Comment by xiaohai12** [comment]
> I will implement it in a few days.
thanks!

**Comment by folkien** [comment]
> > Btw, I already have some blurred data in my dataset.
> 
> The entire image, except for objects, should be very blurry.
> 
> > I will try to set blur=1 to see the result.
> 
> I will implement it in a few days.

Is there an option to set blur= to blur whole image also with objects?
